palette/src/hsl.rs

```diff
@@ -449,6 +455,10 @@ where
     type Scalar = T;
 
     fn saturate(&self, factor: T) -> Hsl<S, T> {
+        self.saturate_fixed((T::one() - self.saturation) * factor)
```

Comment:
Correct me if I'm forgetting some detail, but this should also be `T::max_intensity()`.